controller/konnect/constraints.go

```diff
@@ -0,0 +1,42 @@
+package konnect
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	configurationv1 "github.com/kong/kubernetes-configuration/api/configuration/v1"
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+)
+
+// SupportedKonnectEntityType is an interface that all Konnect entity types
+// must implement.
+type SupportedKonnectEntityType interface {
+	configurationv1alpha1.KongService |
+		configurationv1alpha1.KongRoute |
+		configurationv1.KongConsumer
+	// TODO: add other types
```

Comment:
What about `ConsumerGroup`? Is there anything that prevents us to already add it here?

Reply:
`GetTypeName()` method is missing.

https://github.com/Kong/kubernetes-configuration/pull/15

Reply:
merged. Do you want to bump the `go.mod` and add ConsumerGroup as well or do you prefer doing it in another PR? I'm fine with both

Reply:
Let's do it separately. I also forgot about condition related funcs...